R/manage_deps.R

```diff
@@ -104,18 +104,31 @@ manage_deps <- function(path = ".", profile = "lesson-requirements", snapshot =
 #'   if it's running in an interactive session.
 #' @rdname dependency_management
 #' @export
-update_cache <- function(path = ".", profile = "lesson-requirements", prompt = interactive(), quiet = !prompt, snapshot = TRUE) {
-  path <- root_path(path)
+update_cache <- function(path = ".", profile = "lesson-requirements", prompt = interactive(),
+                         quiet = !prompt, snapshot = TRUE) {
+
   prof <- Sys.getenv("RENV_PROFILE")
   on.exit({
-    invisible(utils::capture.output(renv::deactivate(path), type = "message"))
```

Comment:
Does it affect capturing the output in this case? I don't really understand what `invisible` does though.

Reply:
This bit was moved to `with_renv_factory()` so it will still get executed

- `capture.output()` is used here mainly to suppress the info messages from `renv` (which we don't care about here)
- `inivisble()` returns the output invisibly, in this instance it's again a little "hack" to suppress output